lib/medtronic/processData.js

```diff
@@ -417,23 +425,231 @@ function buildBGRecords(records) {
   return postrecords;
 }
 
+function buildSuspendResumeRecords(records) {
+
+  var suspendResumeRecords = filterHistory([RECORD_TYPES.PUMP_SUSPEND, RECORD_TYPES.PUMP_RESUME], records);
+  var postrecords = [];
+
+  suspendResumeRecords.forEach(function (entry) {
+    console.log('suspend/resume:',entry.jsDate, common.bytes2hex(entry.head), common.bytes2hex(entry.date), common.bytes2hex(entry.body));
+  });
+
+  for(var i=0; i < suspendResumeRecords.length; i++) {
+    var suspendEntry = suspendResumeRecords[i];
+
+    if(suspendEntry.type.value === RECORD_TYPES.PUMP_SUSPEND.value) {
+
+      var suspendedBasal = cfg.builder.makeSuspendBasal()
+        .with_deviceTime(sundial.formatDeviceTime(suspendEntry.jsDate))
+        .set('index', suspendEntry.index);
+      cfg.tzoUtil.fillInUTCInfo(suspendedBasal, suspendEntry.jsDate);
+      postrecords.push(suspendedBasal);
+
+      var suspendResume = cfg.builder.makeDeviceEventSuspendResume()
+        .with_deviceTime(sundial.formatDeviceTime(suspendEntry.jsDate))
+        .with_reason({suspended: 'manual', resumed: 'manual'}) // TODO: get reason
+        .set('index', suspendEntry.index);
+      cfg.tzoUtil.fillInUTCInfo(suspendResume, suspendEntry.jsDate);
+
+      var resumeEntry = suspendResumeRecords[i+1];
+      if(resumeEntry && resumeEntry.type.value === RECORD_TYPES.PUMP_RESUME.value) {
+        i += 1; // increment counter, as we have a matching resume
```

Comment:
When you suspend before uploading, i.e., it's the last record? It's not necessary to suspend for uploading, but possible. I have not seen two suspends in succession, no.